stdlib/test/builtin/test_string.mojo

```diff
@@ -1090,6 +1090,87 @@ def test_indexing():
     assert_equal(a[2], "c")
 
 
+def test_format_args():
+    with assert_raises(contains="Index 1 not in *args"):
+        _ = String("A {0} B {1}").format("First")
+
+    with assert_raises(contains="Index 1 not in *args"):
+        _ = String("A {1} B {0}").format("First")
+
+    with assert_raises(contains="Index 1 not in *args"):
+        _ = String("A {1} B {0}").format()
+
+    with assert_raises(
+        contains="Automatic indexing require more args in *args"
+    ):
+        _ = String("A {} B {}").format("First")
+
+    with assert_raises(
+        contains="Cannot both use manual and automatic indexing"
+    ):
+        _ = String("A {} B {1}").format("First", "Second")
+
+    with assert_raises(contains="Index first not in kwargs"):
+        _ = String("A {first} B {second}").format(1, 2)
+
+    assert_equal(
+        String(" {} , {} {} !").format(
+            "Hello",
+            "Beautiful",
+            "World",
+        ),
+        " Hello , Beautiful World !",
+    )
+
+    assert_equal(String("{{0}}").format("foo"), "{foo}")
```

Comment:
In Python, double braces are an escape sequence for `format`, so `"{{0}}".format("foo")` will actually yield `{0}`. Can we not support that for some reason?

Reply:
I think we can support that easily with small changes

Reply:
:partying_face:  Done but we will need to decomplexify it,
i'm learning `🐍python.format` on the fly and it's relatively complex (for me),
let's continue the hill climb !